src/lib/media-query/match-media.ts

```diff
@@ -51,19 +92,36 @@ export interface MediaQueryList {
  */
 @Injectable()
 export class MatchMedia {
-  protected _registry: Map<string, MediaQueryList>;
+  protected _registry: Map<string, MediaQueryList|ServerMediaQueryList>;
```

Comment:
Instead of using a union type, it would be better to use an interface and an `InjectionToken`. When using a union type, you may end up getting the server-side code on the client

Reply:
Added, great suggestion!